src/test/java/com/google/vinet/servlets/RegistrationServletTest.java

```diff
@@ -0,0 +1,374 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.vinet.servlets;
+
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+
+public class RegistrationServletTest {
+
+  @Mock
+  UserService userService;
+  @Mock
+  HttpServletRequest request;
+  @Mock
+  HttpServletResponse response;
+  @Mock
+  User user;
+  @InjectMocks
+  RegistrationServlet registrationServlet;
+
+  private static final LocalServiceTestHelper helper =
+          new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private static ByteArrayOutputStream errorStream;
+
+  @BeforeEach
+  void injectDependencies() {
+    MockitoAnnotations.openMocks(this);
+  }
+
+  @BeforeEach
+  void redirectErrorStream() {
+    errorStream = new ByteArrayOutputStream();
+    System.setErr(new PrintStream(errorStream));
+  }
+
+  @AfterEach
+  void resetErrorStream() {
+    System.setErr(System.err);
+  }
+
+  @BeforeAll
+  static void setUp() {
+    helper.setUp();
+  }
+
+  @AfterAll
+  static void tearDown() {
+    helper.tearDown();
+  }
+
+  /**
+   * Get any messages that were printed on the error stream
+   *
+   * @return A string containing all errors
+   */
+  private String getErrors() {
+    return errorStream.toString().trim();
+  }
+
+  /**
+   * Sets up the user service mock to have a logged-in and valid user
+   */
+  private void setupUser() {
+    when(userService.isUserLoggedIn()).thenReturn(true);
+    when(userService.getCurrentUser()).thenReturn(user);
+    when(user.getUserId()).thenReturn("anuserid");
+  }
+
+  /**
+   * Sets up an argument captor to get the HTTP status code from the response
+   *
+   * @return The argument captor that will contain the HTTp status code, as an Integer
+   */
+  private ArgumentCaptor<Integer> captureResponseStatus() {
+    ArgumentCaptor<Integer> valueCapture = ArgumentCaptor.forClass(Integer.class);
+    doNothing().when(response).setStatus(valueCapture.capture());
+    return valueCapture;
+  }
```

Comment:
same as before about using `verify()`. Though I know `ArgumentCaptor`'s work, and work well, it is possible to simplify the code somewhat using `verify()` without impacting the functionality.